moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -465,19 +468,71 @@ class OnlineRobot : public Robot
 
 
   /**
+   * @brief execute a given trajectory
+   * @param traj_name - name of trajectory to be executed (must be filled with
+   * prior calls to "addTrajPoint".
+   * @param goal - Joint trajectory goal which is a known 'Plan'
+   * @param collision_check - bool to turn check for collision on\off
+   * @throws <moveit_simple::ExecutionFailureException> (Execution failure)
+   * @throws <moveit_simple::IKFailException> (Conversion to joint trajectory failed)
+   * @throws <std::invalid_argument> (Trajectory "traj_name" not found)
+   * @throws <moveit_simple::CollisionDetected> (One of interpolated point is
+   * in Collision with itself or environment)
+   */
+  void executeKnownPlan(const std::string traj_name, 
+                   control_msgs::FollowJointTrajectoryGoal & goal,
+                   bool collision_check = false);
+
+
+  /**
+   * @brief plan out a given trajectory
+   * @param traj_name - name of trajectory to be executed (must be filled with
+   * prior calls to "addTrajPoint".
+   * @param goal - Joint trajectory goal
+   * @param collision_check - bool to turn check for collision on\off
+   * @throws <moveit_simple::ExecutionFailureException> (Execution failure)
+   * @throws <moveit_simple::IKFailException> (Conversion to joint trajectory failed)
+   * @throws <std::invalid_argument> (Trajectory "traj_name" not found)
+   * @throws <moveit_simple::CollisionDetected> (One of interpolated point is
+   * in Collision with itself or environment)
+   */
+  void plan(const std::string traj_name, 
```

Comment:
Why not return the plan?

Reply:
I was technically sending a reference as an argument. I've modified it so that it returns the plan now.

Reply:
Returning a complex data structure like a plan is not something I would prefer. Instead, we pass an empty plan (type: control_msgs::FollowJointTrajectoryGoal) as a reference argument to the Plan(...) function. This populates the joint trajectory plan which can later be used in the execute(...) function.

Reply:
Is there a reason not to return the data type? The use of this method will read "better" this way?

Reply:
Yeah, it does read better that way. I guess the main reason is that I usually pass complex data types as references for efficiency reasons. But in this case you are right. The method is going to be exposed at the user's end and not at the developer's end. That supersedes my reason for efficiency. I will make the changes.

Reply:
@MauleshSTrivedi: wouldn't RVO take care of this?

Also: I would suggest to avoid this sort of premature optimisation. Methods/functions should return proper types, it makes the intent clear. If/when it turns out that returning 'complex types' (debatable for the plan struct) is a bottleneck, *then* it could make sense to pass in reference objects from caller to callee.

Reply:
yeah actually I forgot about RVO.. Instinctively I usually go for references for these types but your'e right, RVO should take care of this.. :)